pkg/secretless/version_test.go

```diff
@@ -0,0 +1,61 @@
+package secretless
+
+import (
+	"regexp"
+	"testing"
+)
+
+func TestVersionIsPresent(t *testing.T) {
+	version := Version
+
+	if version != "" {
+		return
+	}
+
+	t.Error("Expected Version to be non-empty but got an empty value")
+}
+
+func TestTagIsPresent(t *testing.T) {
+	tag := Tag
+
+	if tag != "" {
+		return
+	}
+
+	t.Error("Expected Tag to be non-empty but got an empty value")
+}
+
+func TestVersionIsCorrectFormat(t *testing.T) {
+	version := Version
+	var validVersionRegex = regexp.MustCompile(`^[0-9]+\.[0-9]+\.[0-9]+$`)
+
+	if validVersionRegex.MatchString(version) {
+		return
+	}
+
+	t.Errorf("Expected Version to be a SemVer string but got '%s'", version)
+}
+
+func TestFullVersionNameIsCorrectFormat(t *testing.T) {
+	fullVersion := FullVersionName
+	var validFullVersionRegex = regexp.MustCompile(`^[0-9]+\.[0-9]+\.[0-9]+-[a-z0-9]+$`)
+
+	if validFullVersionRegex.MatchString(fullVersion) {
+		return
+	}
+
+	t.Errorf("Expected FullVersionName to be a '<SemVer>-<alphanumeric>' string but got '%s'", fullVersion)
+}
+
+// For now we enforce just plain lowercase alphanumerics but we might
+// want to expand this later
+func TestTagIsCorrectFormat(t *testing.T) {
+	tag := Tag
+	var validTagRegex = regexp.MustCompile(`^[a-z0-9]+$`)
+
+	if validTagRegex.MatchString(tag) {
+		return
+	}
+
+	t.Errorf("Expected Tag to be a strict lowercase alphanumeric string but got '%s'", tag)
+}
```

Comment:
Seems like all of these could be tightened up by using assertions.

The irony here is you were the one who lobbied so hard for an assertion lib! :P

Reply:
The assertion lib is there as an option not a requirement - I didn't see these much improved in readability by switching but I'll try to convert these.

Reply:
I stand corrected - readability did improve. Will push an update.